pmm-app/src/pmm-qan/panel/provider/provider.tsx

```diff
@@ -117,83 +116,65 @@ const actions = {
-      to: event.to.utc().format('YYYY-MM-DDTHH:mm:ssZ'),
-      rawTime: {
-        from: event.raw.from,
-        to: event.raw.to,
-      },
-    };
-
-    setContext(newState);
-  }, [panelState]);
-
-  const onTimeRangeChange = useCallback((event) => {
-    const newState = {
+  useEffect(() => {
+    setContext({
       ...panelState,
-      from: event.from.utc().format('YYYY-MM-DDTHH:mm:ssZ'),
-      to: event.to.utc().format('YYYY-MM-DDTHH:mm:ssZ'),
       rawTime: {
-        from: event.raw.from,
-        to: event.raw.to,
+        from: timeRange.raw.from,
+        to: timeRange.raw.to,
       },
-    };
-
-    newState.pageNumber = 1;
-    delete newState.queryId;
-    delete newState.querySelected;
-    setContext(newState);
-  }, [panelState]);
-
-  const dataSource = getDataSourceSrv();
-
-  const templateVariables = (dataSource as any).templateSrv.variables;
-  const { variableSrv } = templateVariables[0];
-  // eslint-disable-next-line no-underscore-dangle
-
-  const [from, setFrom] = useState(rawTime.from);
-  const [to, setTo] = useState(rawTime.to);
+    });
+  }, [timeRange.raw.from, timeRange.raw.to]);
 
   useEffect(() => {
-    // eslint-disable-next-line no-underscore-dangle
-    const handler = variableSrv.dashboard.events.emitter._events['time-range-updated'][0];
-
-    const updateHandler = (event) => {
-      if (from !== event.raw.from || to !== event.raw.to) {
-        onTimeRangeChange(event);
-      } else {
-        onRefresh(event);
-      }
-
-      setFrom(event.raw.from);
-      setTo(event.raw.to);
-      // from = event.raw.from;
-      // to = event.raw.to;
-    };
-
-    handler.fn = updateHandler;
+    refreshGrafanaVariables(panelState);
+  }, [panelState]);
 
-    // eslint-disable-next-line  no-underscore-dangle
-    variableSrv.dashboard.events.emitter._events['time-range-updated'] = [handler];
-  }, [panelState, from, to]);
+  const [from, setFrom] = useState(timeRange.raw.from);
+  const [to, setTo] = useState(timeRange.raw.to);
 
   useEffect(() => {
-    refreshGrafanaVariables(panelState);
-    const newUrl = generateURL(panelState);
-
-    // eslint-disable-next-line no-restricted-globals
-    history.pushState({}, 'test', newUrl);
-  }, [panelState]);
+    if (from === timeRange.raw.from && to === timeRange.raw.to) {
+      const newState = {
+        ...panelState,
+        from: timeRange.from.utc().format('YYYY-MM-DDTHH:mm:ssZ'),
+        to: timeRange.to.utc().format('YYYY-MM-DDTHH:mm:ssZ'),
+        rawTime: {
+          from: timeRange.raw.from,
+          to: timeRange.raw.to,
+        },
+      };
+
+      setContext(newState);
+    } else {
+      const newState = {
+        ...panelState,
+        from: timeRange.from.utc().format('YYYY-MM-DDTHH:mm:ssZ'),
+        to: timeRange.to.utc().format('YYYY-MM-DDTHH:mm:ssZ'),
+        rawTime: {
+          from: timeRange.raw.from,
+          to: timeRange.raw.to,
+        },
+      };
```

Comment:
This can go outside, since it's common between the two branches

Reply:
These changes is for another task, it will fo afrter rebase so don't need to fix it here